test/common/upstream/health_checker_impl_test.cc

```diff
@@ -785,6 +815,29 @@ TEST_F(TcpHealthCheckerImplTest, Timeout) {
   cluster_->runCallbacks({}, removed);
 }
 
+TEST_F(TcpHealthCheckerImplTest, NoData) {
+  InSequence s;
+
+  setupNoData();
+  cluster_->hosts_ = {HostSharedPtr{new HostImpl(
+      cluster_->info_, "", Network::Utility::resolveUrl("tcp://127.0.0.1:80"), false, 1, "")}};
```

Comment:
I think this should be OK since it doesn't actually create a socket, but not sure. In general though would rather not deal with that in this change since the rest of this file looks the same and if there are changes required she can fix them all at the same time.